package/metrics_daemon/src/metrics_daemon.c

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <getopt.h>
+#include <string.h>
+#include <unistd.h>
+#include <ftw.h>
+
+#include <json-c/json.h>
+
+#include <libpiksi/logging.h>
+#include <libpiksi/settings.h>
+
+#define PROGRAM_NAME "metrics_daemon"
+
+#define METRICS_ROOT_DIRECTORY "/tmp/metrics"
+#define METRICS_OUTPUT_FILENAME "/tmp/messaging_statistics-metrics.json"
+#define METRICS_OUTPUT_FULLPATH  METRICS_ROOT_DIRECTORY METRICS_OUTPUT_FILENAME
+#define METRICS_USAGE_UPDATE_INTERVAL (10000u)
+
+const char *metrics_path = METRICS_ROOT_DIRECTORY;
+
+#define MAX_FOLDER_NAME_SZ 64
```

Comment:
Naming/conventions here are "size" for things that include the entire size of an allocated buffer, "length" for strings, which may null terminated.  Since seems to reference to a string length, let's name it `MAX_FOLDER_NAME_LENGTH`.